libraries/bezier-rs/src/subpath/transform.rs

```diff
@@ -75,13 +86,184 @@ impl Subpath {
+				// This is to make sure the the closed loop is correctly represented
+				front.push(front[0]);
+			}
+			front
+		} else {
+			// Determine the subsection of the subpath's manipulator groups that are needed
+			let mut clone = self.manipulator_groups.clone();
+			if self.closed {
+				// Add a duplicate of the first manipulator group to ensure the final closing segment is considered
+				clone.push(clone[0]);
+			}
+
+			// Find the start and end of the new range and consider whether the indices are reversed
+			let range_start = t1_curve_index.min(t2_curve_index);
+			// Add 1 since the drain range is not inclusive
+			// Add 1 again if the corresponding t is not 0 because we want to include the next manipulator group which forms the bezier that this t value is on
+			let range_end = 1 + if are_arguments_reversed {
+				t1_curve_index + ((t1_curve_t != 0.) as usize)
+			} else {
+				t2_curve_index + ((t2_curve_t != 0.) as usize)
+			};
+
+			clone.drain(range_start..range_end.min(clone.len())).collect::<Vec<ManipulatorGroup>>()
+		};
+
+		// Adjust curve indices to match the cloned list
+		if self.closed && are_arguments_reversed {
+			// If trimmed subpath required rotating the manipulator group, adjust the indices to match
+			t2_curve_index = (t2_curve_index + self.len_segments() - t1_curve_index) % self.len_segments();
+			if t2_curve_index == 0 {
+				// If the case is where the start and end are in the same bezier,
+				// change the index to point to the duplicate of this bezier that was pushed to the vector
+				t2_curve_index += self.len_segments();
+			}
+			t1_curve_index = 0;
+		} else {
+			let min_index = t1_curve_index.min(t2_curve_index);
+			t1_curve_index -= min_index;
+			t2_curve_index -= min_index;
+		}
+
+		// Change the representation of the point corresponding to the end point of the subpath
+		// So that we do not need an additional edges case in the later code to handle this point
+		(t1_curve_index, t1_curve_t) = map_index_within_range(t1_curve_index, t1_curve_t, new_manipulator_groups.len() - 1);
+		(t2_curve_index, t2_curve_t) = map_index_within_range(t2_curve_index, t2_curve_t, new_manipulator_groups.len() - 1);
+
+		// Reverse the subpath and update the curve t values to still refer to the same position
+		if !self.closed && are_arguments_reversed {
+			new_manipulator_groups = Subpath::reverse_manipulator_groups(&new_manipulator_groups);
+			t1_curve_t = 1. - t1_curve_t;
+			t2_curve_t = 1. - t2_curve_t;
+		}
+
+		if new_manipulator_groups.len() == 1 {
+			// This case will occur when `t1` and `t2` both represent one of the manipulator group anchors
+			// Add a duplicate manipulator group so that the returned Subpath is still a valid Bezier
+			let mut point = new_manipulator_groups[0];
+			point.in_handle = None;
+			point.out_handle = None;
+			return Subpath {
+				manipulator_groups: vec![point],
+				closed: false,
+			};
+		}
+
+		let len_new_manip_groups = new_manipulator_groups.len();
```

Comment:
The awesome comments above seem to have stopped here, would you mind completing them through the end of the function?